lib/jobs/ucs-discovery.js

```diff
@@ -0,0 +1,169 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di'),
+    urlParse = require('url-parse');
+
+module.exports = UcsDiscoveryJobFactory;
+di.annotate(UcsDiscoveryJobFactory, new di.Provide('Job.Ucs.Discovery'));
+di.annotate(UcsDiscoveryJobFactory, new di.Inject(
+    'Job.Base',
+    'Logger',
+    'Promise',
+    'Assert',
+    'Util',
+    'Services.Waterline',
+    'Services.Encryption',
+    '_',
+    'JobUtils.UcsTool'
+));
+
+function UcsDiscoveryJobFactory(
+    BaseJob,
+    Logger,
+    Promise,
+    assert,
+    util,
+    waterline,
+    encryption,
+    _,
+    UcsTool
+) {
+    var logger = Logger.initialize(UcsDiscoveryJobFactory);
+
+    /**
+     * @param {Object} options task options object
+     * @param {Object} context graph context object
+     * @param {String} taskId running task identifier
+     * @constructor
+     */
+    function UcsDiscoveryJob(options, context, taskId) {
+        UcsDiscoveryJob.super_.call(this,
+                                   logger,
+                                   options,
+                                   context,
+                                   taskId);
+
+        assert.object(this.options);
+        assert.string(this.options.uri);
+        var parse = urlParse(this.options.uri);
+        var protocol = parse.protocol.replace(':','').trim();
+        this.settings = {
+            uri: parse.href,
+            host: parse.host.split(':')[0],
+            root: parse.pathname,
+            port: parse.port,
+            protocol: protocol,
+            username: this.options.username,
+            password: this.options.password,
+            ucs: this.options.ucs,
+            verifySSL: this.options.verifySSL || false
+        };
+        this.ucs = new UcsTool();
+        this.ucs.settings = this.settings;
+    }
+    
+    util.inherits(UcsDiscoveryJob, BaseJob);
+
+    /**
+     * @memberOf UcsDiscoveryJob
+     */
+    UcsDiscoveryJob.prototype._run = function() {
+        var self = this;
+        
+        return self.getRoot()
+            .then(function(root) {
+                return [ root, self.createRackmounts(root) ];
+            })
+            .then(function() {
+                self._done();
+            })
+            .catch(function(err) {
+                self._done(err);
+            });
+    };
+
+    /**
+     * @function logIn
+     * @description Log into the UCS microservice
+     */
+    UcsDiscoveryJob.prototype.logIn = function () {
+        var username = this.settings.username;
```

Comment:
@tannoa2 Not sure if these are checked elsewhere (schema maybe?) but it would probably make sense to add asserts to make sure these settings have the expected type (e.g. `assert.string(username)`).  This applies to a few of the functions in this file.